src/Raven.Server/Documents/Handlers/AI/Agents/ConversationDocument.cs

```diff
@@ -39,14 +39,16 @@ public void Initialize(JsonOperationContext context, AiAgentConfiguration config
         if (Messages.Count > 0)
             throw new InvalidOperationException("conversation document is already initialized. Cannot re-initialize.");
 
-        foreach (var parameter in configuration.Parameters)
+        var relevantParameters = configuration.Parameters.Where(p => p.SendToModel != false).ToList();
```

Comment:
Let's avoid using Linq here, do the filtering in the for loop.
You are allocating a lot here without really needing to